### PR TITLE
Stop double-submit of set password form

### PR DESCRIPTION
**Pull request by tgolen**
### Details
When pressing enter, the form was being submit twice. Once for the text input and one for the submit button. The multiple submits was causing `validateAndSubmitForm()` to be ran multiple times, and the second time it ran, it would have `this.props.userSignUp.authToken` from the first request. This caused `changePasswordAndSignIn()` to be triggered from an invalid state and put the UI into a position where it couldn't fetch all the reports. Since the reports couldn't fully load, it was stuck at a loading spinner.

### Fixed Issues
$ https://github.com/Expensify/App/issues/9058

### Tests
For local tests, you'll need to be able to get the validate code for a new account.
1. Launch NewDot and make sure you're signed out
2. Sign in with a new email address to create a new account
3. From the CLI run `../script/sql.sh`
4. Run the query `SELECT  FROM notifications ORDER BY created DESC LIMIT 0,1;`
5. Look for the `accountID` and `validateCode`
6. Go back to NewDot and modify the URL to go to `http://localhost:8080/setpassword/<accountID>/<validateCode>` (use the values from the query)
7. Enter a valid password and press the ENTER key
8. Verify that you are taken into the app and everything loads
9. Repeat the entire process with another new user, this time, instead of pressing the ENTER key on step 7, click the submit button

- [ ] Verify that no errors appear in the JS console

### QA Steps
1. Go to URL https://staging.new.expensify.com/
2. In the email field, enter a email address for a new account and press continue
3. Navigate to the email inbox of the account and open the validation email
4. Copy the link and change it to staging
5. Set a password that follows the password requirements
6. Submit the form by pressing the ENTER key
7. Verify the app loads everything properly
8. Repeat all the steps but for step 6, click the submit button on the form

- [ ] Verify that no errors appear in the JS console

### Screenshots
This but only effects web because that is the only way to validate an account.

#### Web

https://user-images.githubusercontent.com/1228807/173594369-4f78fd3c-c737-4035-a6a3-99303f1eee81.mp4

https://user-images.githubusercontent.com/1228807/173593719-8b1a2442-3e93-43fe-820c-acd384b9a4d3.mp4

### CODE REVIEW

File: src/pages/SetPasswordPage.js [hunk 1]
```diff
@@ -126,7 +126,6 @@ class SetPasswordPage extends Component {
                             password={this.state.password}
                             updatePassword={password => this.setState({password})}
                             updateIsFormValid={isValid => this.setState({isFormValid: isValid})}
-                            onSubmitEditing={this.validateAndSubmitForm}
```
[the remark this answers is not in the document]
Comment (tgolen) [reply src/pages/SetPasswordPage.js:129]: Ah, good to know. I'll put a no-op instead then.

**Comment by tgolen** [comment]
Updated

**Review by aldo-expensify — APPROVED** [review]

**Comment by aldo-expensify** [comment]
@tgolen I just noticed that this PR included changes on the `package-lock.json`, do we want to reverse?

**Comment by tgolen** [comment]
@aldo-expensify ah, phooey. That wasn't intended to get through. I'm sure it's fine (for the sake of production and for NPM packages), but I'm not sure what havok this will cause to other devs. Once we are all on M1s it will probably self-resolve. I think it's OK to lave as-is for now though.

**Pull request merged**